api/download_ics.go

```diff
@@ -0,0 +1,104 @@
+package api
+
+import (
+	"errors"
+	"github.com/getsentry/sentry-go"
+	"github.com/gin-gonic/gin"
+	"github.com/joschahenningsen/TUM-Live/dao"
+	"github.com/joschahenningsen/TUM-Live/model"
+	"github.com/joschahenningsen/TUM-Live/tools"
+	"html/template"
+	"net/http"
+	"strconv"
+)
+
+func configGinDownloadICSRouter(router *gin.Engine) {
+	router.GET("/api/download_ics/:year/:term/:slug", downloadICS)
```

Comment:
`/api/course/:year/:term/:slug/events.ics` would be a little nicer semantically

Reply:
You can group this and attatch the middleware `tools.InitCourse` like here

https://github.com/joschahenningsen/TUM-Live/blob/1f90e3841bfa43f7731b1b61e9103c8d28329738/api/lecture_halls.go#L35

This way you can get the course from the context in the route instead of manually using the params. (However, the middleware checks if the user requesting the file is allowed to see the course which makes it infeasible if we want calendar software to be able to fetch the file)

Reply:
I am already checking if the User is logged in with
```
tumLiveContext := foundContext.(tools.TUMLiveContext)
  if tumLiveContext.User == nil {
    c.AbortWithStatus(http.StatusForbidden)
    return
}
```
Should I remove this to make sure calendar software can fetch the file? Or should I remove this and do as you suggested (attach `tools.InitCourse`)

Reply:
Lecture dates are public information in tumonline anyway, let's remove the user check as well and keep the course fetching as is :) If we need some kind of authentication here in the future we'd need to build a token based system

Reply:
Alright, I am updating the code!